sound/soc/sof/topology.c

```diff
@@ -2152,10 +2233,75 @@ static int sof_link_load(struct snd_soc_component *scomp, int index,
 	return 0;
 }
 
+static int sof_link_hda_unload(struct snd_sof_dev *sdev,
+			       struct snd_soc_dai_link *link)
+{
+	struct snd_soc_dai_link_component dai_component = {0};
+	struct snd_soc_dai *dai;
+	u32 tx_num = 0;
+	u32 rx_num = 0;
+	int ret = 0;
+
+	dai_component.dai_name = link->cpu_dai_name;
+	dai = snd_soc_find_dai(&dai_component);
+	if (!dai) {
+		dev_err(sdev->dev, "failed to find dai %s", dai->name);
+		return -EINVAL;
+	}
+
+	if (link->dpcm_playback)
+		tx_num = 1;
+
+	if (link->dpcm_capture)
+		rx_num = 1;
+
+	ret = snd_soc_dai_set_channel_map(dai, tx_num, NULL,
+					  rx_num, NULL);
```

Comment:
You could just do
ret = snd_soc_dai_set_channel_map(dai, link->dpcm_playback, NULL,
					  link->dpcm_capture, NULL);
and eliminate rx_num and tx_num completely in this function

Reply:
link->dpcm_playback is defined as bool variable, so I don't do like this.